src/Operation/Aggregate.php

```diff
@@ -300,6 +300,10 @@ public function getCommandDocument()
     {
         $cmd = $this->createCommandDocument();
 
+        // The 'explain' option is incompatible with the explain command
+        unset($cmd['explain']);
```

Comment:
I get the following error if the option `explain` is set, even if it's `false`.

> MongoDB\Driver\Exception\CommandException: The 'explain' option is illegal when a explain verbosity is also provided

Reply:
I'm OK relying on the server to raise an error here if you'd like to remove this.

I'd argue this is fundementally different from intentionally omitting `writeConcern` ([PHPLIB-802](https://jira.mongodb.org/browse/PHPLIB-802)) because that option is easily inherited from a parent context.

If a user encounters the error you highlighted above, it's because they've explicitly crafted an Aggregate operation with an `explain` option, and then decided to pass that to `Collection::explain()`.